Client/mods/deathmatch/logic/CClientPed.cpp

```diff
@@ -4041,6 +4041,13 @@ void CClientPed::ReCreateGameEntity()
 
 void CClientPed::ModelRequestCallback(CModelInfo* pModelInfo)
 {
+    // The model loading may take a while and there's a chance of ped being moved to other dimension.
+    if (!IsVisibleInAllDimensions() && GetDimension() != m_pStreamer->GetDimension())
```

Comment:
```c++
if (!ShouldBeVisibleInCurrentDimension())
```
And you can keep `m_pStreamer` private

Reply:
I don't think that a dedicated method is appropriate here. `m_pStreamer` is absolutely okay here.